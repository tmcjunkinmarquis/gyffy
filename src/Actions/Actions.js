export const inputLocation = (location)=>({
  type: 'INPUT_LOCATION',
  location
});

export const loadStations = (stations)=>({
  type: 'LOAD_STATIONS',
  stations
});