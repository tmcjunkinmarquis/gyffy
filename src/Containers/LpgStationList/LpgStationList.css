.wrapper {
  background-color: #E8EEF1;
  background-color: 100vh;
  overflow-x: hidden;
  overflow-y: hidden;
  text-shadow: none;
  color: gray;
}

.fuel-station-list {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
}

.fuel-stations-header {
  margin-left: 20px;
  color: black;
  text-shadow: none;
}

span {
  color: darkslateblue;
  white-space: pre;
}

.filter-type {
  color: orange;
}