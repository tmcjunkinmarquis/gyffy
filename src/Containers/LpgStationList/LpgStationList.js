import React, { Component } from 'react';
import { connect } from 'react-redux';
import './LpgStationList.css';
import StationCard from '../../Components/StationCard/StationCard';
import PropTypes from 'prop-types';
import { setFilter } from '../../Actions/Actions';
import { Link } from 'react-router-dom';

class LpgStationList extends Component {

  componentDidMount() {
    this.props.handleFilterClick('SHOW_LPG');
  }

  filterToReturnLpgStations = () => {
    const filteredStations = this.props.stations.filter((station) => {
      return station.fuelType === 'LPG';
    });
    return filteredStations.map((station) => {
      return <Link
        to={`/details/${station.id}`}
        key={station.id}>
        <StationCard
          station={station}
          fuelType={station.fuelType} />
      </Link>;
    });
  };

  fillTheSpan = ()=>{
    return 'LPG   '
  }

  render() {
    return (
      <div className="wrapper">
        <h3 className="fuel-stations-header"><span className="filter-type">{this.fillTheSpan()}</span>STATIONS
          <span >   -   {this.props.location}</span>
        </h3> 
        <div className="fuel-station-list">
          {this.filterToReturnLpgStations()}
        </div>
      </div>
    );
  }
}

export const mapStateToProps = (state) => ({
  location: state.location,
  stations: state.stations,
  filter: state.filter
});

export const mapDispatchToProps = (dispatch) => ({
  handleFilterClick: (filter) => dispatch(setFilter(filter))
});

export default connect(mapStateToProps, mapDispatchToProps)(LpgStationList);

LpgStationList.propTypes = {
  stations: PropTypes.array,
  handleFilterClick: PropTypes.func,
  location: PropTypes.string,
  filter: PropTypes.string
};