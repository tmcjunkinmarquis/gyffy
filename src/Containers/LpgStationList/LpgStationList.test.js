import React from 'react';
import { LpgStationList, mapStateToProps, mapDispatchToProps } from './LpgStationList';
import { shallow } from 'enzyme';
import { setFilter } from '../../Actions/Actions';

describe('LpgStationList Container', () => {

  describe('LpgStationList Component', () => {
    let wrapper;
    let mockProps;
    
    beforeEach(()=>{
      const mockHandleFilterClick = jest.fn();
      mockProps = {
        stations: [{ name: 'bubbas' }],
        location: '61615',
        filter: 'SHOW_LPG'
      };
      wrapper = shallow(
        <LpgStationList 
          {...mockProps}
          handleFilterClick={mockHandleFilterClick}
        />);
    });

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });


  describe('mapStateToProps', () => {
    let wrapper;
    let mockProps;
    const mockHandleFilterClick = jest.fn();
    beforeEach(() => {
      mockProps = {
        stations: [{ name: 'bubbas' }],
        location: '61615',
        filter: 'SHOW_ELEC'
      };
      wrapper = shallow(
        <LpgStationList
          {...mockProps}
          handleFilterClick={mockHandleFilterClick}
        />);
    });

    it('should return an object with keys location, stations, and filter', () => {
      const mockState = {
        location: '61615',
        stations: [{ name: 'bubbas' }],
        filter: 'SHOW_LPG'
      };
      const expected = {
        location: '61615',
        stations: [{ name: 'bubbas' }],
        filter: 'SHOW_LPG'
      };
      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected);
    });
  }); 

  describe('mapDispatchToProps', () => {
    let wrapper;
    let mockProps;
    const mockHandleFilterClick = jest.fn();

    beforeEach(() => {
      mockProps = {
        stations: [{ name: 'bubbas' }],
        location: '61615',
        filter: 'SHOW_LPG'
      };
      wrapper = shallow(
        <LpgStationList
          handleFilterClick={mockHandleFilterClick}
          {...mockProps}
        />);
    });

    it('calls dispatch with a setFilter action when handleFilterClick is called', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = setFilter('SHOW_LPG');
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.handleFilterClick('SHOW_LPG');

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  }); 
  
});