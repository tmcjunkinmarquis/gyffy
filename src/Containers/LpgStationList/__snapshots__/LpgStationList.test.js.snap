// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LpgStationList Container LpgStationList Component should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LpgStationList
    filter="SHOW_LPG"
    handleFilterClick={[Function]}
    location="61615"
    stations={
        Array [
            Object {
              "name": "bubbas",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3
          className="fuel-stations-header"
>
          <span
                    className="filter-type"
          >
                    LPG   
          </span>
          STATIONS
          <span>
                       -   
                    61615
          </span>
</h3>,
        <div
          className="fuel-station-list"
>
          
</div>,
      ],
      "className": "wrapper",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="filter-type"
>
              LPG   
</span>,
            "STATIONS",
            <span>
                 -   
              61615
</span>,
          ],
          "className": "fuel-stations-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "LPG   ",
              "className": "filter-type",
            },
            "ref": null,
            "rendered": "LPG   ",
            "type": "span",
          },
          "STATIONS",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "   -   ",
                "61615",
              ],
            },
            "ref": null,
            "rendered": Array [
              "   -   ",
              "61615",
            ],
            "type": "span",
          },
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
          "className": "fuel-station-list",
        },
        "ref": null,
        "rendered": Array [],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            className="fuel-stations-header"
>
            <span
                        className="filter-type"
            >
                        LPG   
            </span>
            STATIONS
            <span>
                           -   
                        61615
            </span>
</h3>,
          <div
            className="fuel-station-list"
>
            
</div>,
        ],
        "className": "wrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="filter-type"
>
                LPG   
</span>,
              "STATIONS",
              <span>
                   -   
                61615
</span>,
            ],
            "className": "fuel-stations-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "LPG   ",
                "className": "filter-type",
              },
              "ref": null,
              "rendered": "LPG   ",
              "type": "span",
            },
            "STATIONS",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "   -   ",
                  "61615",
                ],
              },
              "ref": null,
              "rendered": Array [
                "   -   ",
                "61615",
              ],
              "type": "span",
            },
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "fuel-station-list",
          },
          "ref": null,
          "rendered": Array [],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
